Dockerfile

```diff
@@ -9,6 +9,34 @@ RUN mkdir /data/TestXBE
 RUN /usr/src/nxdk/docker_entry.sh make -C /test-xbe
 RUN cp /test-xbe/tester.iso /data/TestXBE/
 
+# The nxdk_pgraph_tests includes its own copy of the nxdk which needs more build
+# infrastructure than the nxdk-runbase used for other tests.
+FROM ubuntu:20.04 AS pgraph-buildbase
+ENV DEBIAN_FRONTEND noninteractive
+RUN set -xe; \
+    apt-get -qy update \
+    && apt-get -qy install \
+        bison \
+        clang \
+        cmake \
+        flex \
+        lld \
+        llvm \
+        make
+
+FROM pgraph-buildbase AS pgraph-data
+RUN mkdir -p /data/TestNXDKPgraphTests
+COPY test-pgraph /test-pgraph
+RUN make -C /test-pgraph/nxdk_pgraph_tests \
+    AUTORUN_IMMEDIATELY=y \
+    ENABLE_SHUTDOWN=y \
+    FALLBACK_OUTPUT_ROOT_PATH="c:" \
+    RUNTIME_CONFIG_PATH="c:/pgraph_tests.cnf" \
+    CC=clang CXX=clang++
```

Comment:
Let's `-j$(numproc)` to speed up the build

Reply:
Done.